client/scripts/app.ts

```diff
@@ -819,48 +825,17 @@ Promise.all([
   initAppState(true, customDomain).then(async () => {
     // setup notifications and websocket if not already set up
     if (!app.socket) {
-      let jwt;
-      // refresh notifications once
       if (app.loginState === LoginState.LoggedIn) {
+        // refresh notifications once
         app.user.notifications.refresh().then(() => m.redraw());
-        jwt = app.user.jwt;
-      }
-      // grab discussion drafts
-      if (app.loginState === LoginState.LoggedIn) {
+        // grab all discussion drafts
         app.user.discussionDrafts.refreshAll().then(() => m.redraw());
+        app.socket = new WebSocketController(app.user.jwt);
```

Comment:
Any reason we construct a WebSocketController in `initAppState` and then again in the callback?

Reply:
Good catch - refactored the logic and removed the duplicate init for the socket.